stock_barcodes/wizard/stock_barcodes_read_picking.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2019 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from odoo import api, _, fields, models
+from odoo.fields import first
+from odoo.addons import decimal_precision as dp
+
+_logger = logging.getLogger(__name__)
+
+
+class WizStockBarcodesReadPicking(models.TransientModel):
+    _name = 'wiz.stock.barcodes.read.picking'
+    _inherit = 'wiz.stock.barcodes.read'
+    _description = 'Wizard to read barcode on picking'
+
+    picking_id = fields.Many2one(
+        comodel_name='stock.picking',
+        string='Picking',
+        readonly=True,
+    )
+    candidate_picking_ids = fields.One2many(
+        comodel_name='wiz.candidate.picking',
+        inverse_name='wiz_barcode_id',
+        string='Candidate pickings',
+        readonly=True,
+    )
+    picking_product_qty = fields.Float(
+        string='Picking quantities',
+        digits=dp.get_precision('Product Unit of Measure'),
+        readonly=True,
+    )
+    picking_type_code = fields.Selection([
+        ('incoming', 'Vendors'),
+        ('outgoing', 'Customers'),
+        ('internal', 'Internal'),
+    ], 'Type of Operation')
+
+    def name_get(self):
+        return [
+            (rec.id, '{} - {} - {}'.format(
+                _('Barcode reader'),
+                rec.picking_id.name, self.env.user.name)) for rec in self]
+
+    def _set_default_picking(self):
+        picking_id = self.env.context.get('default_picking_id', False)
+        if picking_id:
+            self._set_candidate_pickings(
+                self.env['stock.picking'].browse(picking_id))
+
+    @api.model
+    def create(self, vals):
+        # When user click any view button the wizard record is create and the
+        # picking candidates have been lost, so we need set it.
+        wiz = super().create(vals)
+        wiz._set_default_picking()
+        return wiz
+
+    @api.onchange('picking_id')
+    def onchange_picking_id(self):
+        # Add to candidate pickings the default picking. We are in a wizard
+        # view, so for create a candidate picking with the same default picking
+        # we need create it in this onchange
+        self._set_default_picking()
```

Comment:
An onchange **musn't** create records. You should switch to a compute virtual many2many field.